poetry/console/commands/init.py

```diff
@@ -67,9 +67,20 @@ def handle(self):
         from poetry.utils._compat import Path
         from poetry.utils.env import SystemEnv
 
-        if (Path.cwd() / "pyproject.toml").exists():
-            self.line("<error>A pyproject.toml file already exists.</error>")
-            return 1
+        original_toml = PyProjectTOML(Path.cwd() / "pyproject.toml")
+
+        if original_toml.file.exists():
+            if original_toml.is_poetry_project():
```

Comment:
You can combine this, is project should check if the file exists.

Reply:
Yes, but for the check if a build-system is defined, I have to check if the file exists first anyway. So I found it cleaner to read if I first check if the file exists and then include any test I like in this block.